src/templates/index.html

```diff
@@ -118,7 +181,155 @@ <h2>Support & Additional Resources</h2>
+	         var row= jsonData[i];
+            
+			 var lat=row['Latitude'];
+             
+	         var lng=row['Longitude'];
+             
+			 var hoseLength=row['Hose Length'];
+             
+			 var transportMode=row['Transport Mode'];
+             
+			 var pointType=row['Point Type'];
+             
+			 
+			var popupInput = L.popup() // Create and display the popup object on the map;
+            .setLatLng({ 'lat': lat, 'lng': lng })
+            .openOn(map);
+            
+
+			var redFireIcon = new L.Icon({
+                iconUrl: 'https://raw.githubusercontent.com/pointhi/leaflet-color-markers/master/img/marker-icon-2x-red.png',
+                shadowUrl: 'https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png',
+                iconSize: [25, 41],      // Adjust the icon size as needed
+                iconAnchor: [12, 41],    // Adjust the icon anchor point as needed
+                popupAnchor: [1, -34],   // Adjust the popup anchor point as needed
+                shadowSize: [41, 41]     // Adjust the shadow size as needed
+            });
+
+            if (hoseLength !== '') {
+
+                //add the marker object to the map
+               // var marker = L.marker([lat, lng]).addTo(map);
+
+                // Check if the point type is "fire" and use the custom red marker icon
+                if (pointType === 'Fire') {
+                    var marker = L.marker([lat, lng], { icon: redFireIcon }).addTo(map);
+                    
+                } else {
+                    // For other point types, use the default icon or a different one
+                    var marker = L.marker([lat, lng]).addTo(map);
+                    
+                }
+
+                // add the point to the points dictionary and get the generated ID of the point. 
+                id = addPointDataToDictionary(
+                    lat,
+                    lng,
+                    hoseLength,
+                    transportMode,
+                    marker,
+                    pointType // Include point_type here
+                );
+                
+                //now create a new popup which just display the input data
+                var popupContentDisplay =
+                    `<body>
+                        <p>Hose Length: ${hoseLength}</p>
+                        <p>Mode: ${transportMode}</p>
+                        <p>Point Type: ${pointType}</p>
+                        <button 
+                            id="popupDeletePointButton"
+                            onclick='handleDeletePoint(${id})'
+                        >
+                            Delete Point
+                        </button>
+                    </body>`;
+
+                // Bind the display popup to the marker so it displays when the marker is clicked. 
+                marker.bindPopup(popupContentDisplay)
+                map.closePopup(popupInput);
+            }
+        };
+    }
```

Comment:
This section looks like a restatement of code we already have defined in the functions below for adding points. Question: can you not re-use the existing functions that already do things like add points instead of repeat code blocks? if so, why not? avoiding reusing those functions will mean we have to maintain two ways of adding points ongoing, which is undesirable compared to only have one function that controls the process.

Reply:
There is a slight difference in reading the points from UI one after the other vs reading it from excel. In order to not effect anything from the existing code. I preferred creating new function to handle excel inputs, later redirected to process the output in the existing way.

We can discuss in more detail. I will put it on the agenda for our next meeting.